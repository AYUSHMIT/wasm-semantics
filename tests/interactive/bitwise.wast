i32.const 2
i32.const 1
i32.shl
i32.const 4
i32.eq

i32.const 2
i32.const #pow31 +Int 1
i32.shl
i32.const 4
i32.eq

i32.const 2
i32.const #pow31
i32.shr_u
i32.const 2 ^Int 29
i32.eq

i32.const 2
i32.const 2
i32.shr_u
i32.const 0
i32.eq

i32.const 1
i32.const #pow32 -Int 2
i32.shr_s
i32.const #pow32 -Int 1
i32.eq

i32.const 2
i32.const 2
i32.shr_s
i32.const 0
i32.eq
