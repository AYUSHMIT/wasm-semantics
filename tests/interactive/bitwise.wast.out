<generatedTop>
  <k>
    .
  </k>
  <stack>
    < i32 > 1 : < i32 > 1 : < i32 > 1 : < i32 > 1 : < i32 > 1 : < i32 > 1 : .Stack
  </stack>
</generatedTop>
